### PR TITLE
[USD-231][UST-229][UST-256] Remove the cache and post filter content and keyword selection

### PR DESCRIPTION
**Pull request by Lisooo790926**
Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary / Details
1. Removed the post cache, ((( please review my comment in this ticket(#520), I believe the fetching post behavior will be bottleneck after removing the cache. 
2. Removed the post filter and fixed related test
3. Added keyword selection ((( this is related above fetching post behavior, add search by keyword, and it should also impact post fetching 

## Linked Issue
#520 
#544 
#546 

## Impacted Areas
post/account/vote/comment, all behaviors that require post fetching are impacted.

## Tests
Minor add tests for keyword selection, which should not impact current tests.

## Verification Steps
Try keyword selection, the query response should have related content

## Todo
-   [x] Removed Post Cache and rename the method `updateOrder`  to  `fetchOrderedPosts`
-   [x] Removed all parts where used this method 
-   [x] Removed filterPostContent method and fix related tests
-   [x] Added keywords as param in post.router and added the related test

## Checklist
-   [x] All new and existing tests pass

### CODE REVIEW

File: packages/relay/test/post.test.ts [hunk 1]
```diff
@@ -318,6 +317,21 @@ describe('POST /post', function () {
         expect(offChainPost.status).equal(0)
     })
 
+    it('should fetch posts by keyword', async function () {
+        let posts = await express
+            .get(`/api/post?page=1&keyword=content`)
```
Comment (ryanycw) [inline packages/relay/test/post.test.ts:322]:
```suggestion
            .get(`/api/post?q=content`)
```

Comment (ryanycw) [reply packages/relay/test/post.test.ts:322]: page is default to 1 should be able to skip it

File: packages/relay/src/services/PostService.ts [hunk 2]
```diff
@@ -183,11 +150,11 @@ export class PostService {
 
         return await Promise.all(
             posts.map(async (post) => {
+                // TODO should remove loop sql query and replace it as join
```
Comment (ryanycw) [inline packages/relay/src/services/PostService.ts:153]: Note: Issue to be opened.

File: packages/relay/test/report.test.ts [hunk 3]
```diff
@@ -190,7 +191,7 @@ describe('POST /api/report', function () {
             (post) => post.postId === postId
         )
         expect(reportedPost).to.exist
-        expect(reportedPost).to.not.have.property('content')
+        expect(reportedPost).to.have.property('content', 'test content')
```
Comment (ryanycw) [inline packages/relay/test/report.test.ts:194]: I think this is a typo?

File: packages/relay/test/report.test.ts [hunk 4]
```diff
@@ -979,7 +980,7 @@ describe('POST /api/report', function () {
         await express.get(`/api/post/${report.objectId}`).then((res) => {
             const curPost = res.body as Post
             expect(curPost.status).to.equal(PostStatus.DISAGREED)
-            expect(curPost).to.not.have.property('content')
+            expect(curPost.content).equal('test content')
```
Comment (ryanycw) [inline packages/relay/test/report.test.ts:983]: So is here?

**Review by ryanycw — CHANGES_REQUESTED** [review]
Thank you in general I believe all issues are resolved!!

File: packages/relay/test/report.test.ts (hunk 3 above)
Comment (Lisooo790926) [reply packages/relay/test/report.test.ts:194]: this is a response, it should be content

**Review by ryanycw — APPROVED** [review]
Everything looks good!! Thank you for your work

File: packages/relay/src/services/PostService.ts (hunk 2 above)
Comment (ryanycw) [reply packages/relay/src/services/PostService.ts:153]: @Lisooo790926 Can you create a new issue for this, rebase the PR and feel free to merge it.

**Pull request merged**